File: app/src/main/java/com/example/administrator/classromapplication/view/ui/LoginActivity.java
package com.example.administrator.classromapplication.view.ui;

import android.view.View;

import com.example.administrator.classromapplication.LoginUtil;
import com.example.administrator.classromapplication.R;
import com.example.administrator.classromapplication.databinding.ActivityLoginBinding;
import com.example.administrator.classromapplication.viewmodel.UserViewModel;


public class LoginActivity extends BaseActivity<ActivityLoginBinding, UserViewModel> {


    @Override
    protected void initListener() {
        binding.btnLogin.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(final View v) {
                loginToBmob();
            }
        });
        binding.tvRegistered.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                toRegisterAct();
            }
        });
    }

    private void toRegisterAct() {
        startActForResult(RegisteredActivity.class, null, 0);
    }

    @Override
    protected int getLayoutId() {
        return R.layout.activity_login;
    }


    private void loginToBmob() {
        LoginUtil.login(viewModel.userName, viewModel.pwd, this);
    }


    @Override
    protected void initView() {
        binding.setLogin(viewModel);
    }

    @Override
    protected UserViewModel getViewModel() {
        UserViewModel model = new UserViewModel();
        binding.setLogin(model);
        return model;
    }
}

File: app/src/main/java/com/example/administrator/classromapplication/view/ui/LoginActivity.kt
package com.example.administrator.classromapplication.view.ui

import com.example.administrator.classromapplication.LoginUtil
import com.example.administrator.classromapplication.R
import com.example.administrator.classromapplication.databinding.ActivityLoginBinding
import com.example.administrator.classromapplication.viewmodel.UserViewModel
import kotlinx.android.synthetic.main.activity_login.*

/**
 * Created by Ervin on 2018/1/10.
 */
class LoginActivity : BaseActivity<ActivityLoginBinding, UserViewModel>() {
    override fun getViewModel(): UserViewModel {
        val model = UserViewModel()
        binding.login = model
        return model
    }

    override fun initListener() {
        btn_login.setOnClickListener {
            LoginUtil.login(viewModel.userName, viewModel.pwd, this)
        }
        tv_registered.setOnClickListener {
            startActForResult(RegisteredActivity::class.java, null, 0)
        }
    }

    override fun getLayoutId(): Int {
        return R.layout.activity_login
    }
}